Mathlib/Data/Fin/Tuple/Basic.lean

```diff
@@ -724,9 +724,44 @@ def succAboveCases {α : Fin (n + 1) → Sort u} (i : Fin (n + 1)) (x : α i)
     else @Eq.recOn _ _ (fun x _ ↦ α x) _ (succAbove_pred_of_lt _ _ <|
     (Fin.lt_or_lt_of_ne hj).resolve_left hlt) (p _)
 
-theorem forall_iff_succAbove {p : Fin (n + 1) → Prop} (i : Fin (n + 1)) :
-    (∀ j, p j) ↔ p i ∧ ∀ j, p (i.succAbove j) :=
-  ⟨fun h ↦ ⟨h _, fun _ ↦ h _⟩, fun h ↦ succAboveCases i h.1 h.2⟩
+/-- This is a duplicate of `Fin.forall_fin_succ` in Core. We should upstream the name change. -/
+lemma forall_iff_succ {P : Fin (n + 1) → Prop} : (∀ i, P i) ↔ P 0 ∧ ∀ i, P (succ i) :=
+  ⟨fun h ↦ ⟨h _, fun _ ↦ h _⟩, fun h ↦ cases h.1 h.2⟩
+
+/-- This is a duplicate of `Fin.exists_fin_succ` in Core. We should upstream the name change. -/
+lemma exists_iff_succ {P : Fin (n + 1) → Prop} : (∃ i, P i) ↔ P 0 ∨ ∃ i, P (succ i) where
+  mp := by
```

Comment:
I think we should give an alias here - together with the same comment - rather than re-proving the lemma. Same above.